open-machine-learning-jupyter-book/ml-advanced/clustering/k-means-clustering.md

```diff
@@ -0,0 +1,280 @@
+---
+jupytext:
+  cell_metadata_filter: -all
+  formats: md:myst
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.11.5
+kernelspec:
+  display_name: Python 3
+  language: python
+  name: python3
+---
+
+# K-Means clustering
+
+In this section, you will learn how to create clusters using Scikit-learn and the Nigerian music dataset you imported earlier. We will cover the basics of K-Means for Clustering. Keep in mind that, as you learned in the earlier section, there are many ways to work with clusters and the method you use depends on your data. We will try K-Means as it's the most common clustering technique. Let's get started!
+
+Terms you will learn about:
+
+- Silhouette scoring
+- Elbow method
+- Inertia
+- Variance 
+
+## Introduction
+
+K-Means Clustering is a method derived from the domain of signal processing. It is used to divide and partition groups of data into 'k' clusters using a series of observations. Each observation works to group a given datapoint closest to its nearest 'mean', or the center point of a cluster.
+
+The clusters can be visualized as Voronoi diagrams, which include a point (or 'seed') and its corresponding region. 
+
+```{figure} ../../../images/clustering/voronoi.png
+---
+name: voronoi diagram
+---
+voronoi diagram
+infographic by Jen Looper
+```
+
+The K-Means clustering process executes in a three-step process):
+
+- The algorithm selects k-number of center points by sampling from the dataset. After this, it loops:
+
+1. It assigns each sample to the nearest centroid.
+2. It creates new centroids by taking the mean value of all of the samples assigned to the previous centroids.
+3. Then, it calculates the difference between the new and old centroids and repeats until the centroids are stabilized.
+
+One drawback of using K-Means includes the fact that you will need to establish 'k', that is the number of centroids. Fortunately the  'elbow method' helps to estimate a good starting value for 'k'. You'll try it in a minute.
+
+## Prerequisite
+
+You will work in this section's _notebook.ipynb_ file that includes the data import and preliminary cleaning you did in the last section.
+
+## Exercise - preparation
+
+Start by taking another look at the songs data.
+
+```{code-cell}
+!pip install seaborn
```

Comment:
same as above. We may not need this line. But you can keep it as it actually does not take effect.